includes/class-wc-payment-tokens.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * WooCommerce Payment Tokens
+ *
+ * An API for storing and managing tokens for gateways and customers.
+ *
+ * @class 		WC_Payment_Tokens
+ * @since		2.6.0
+ * @package		WooCommerce/Classes
+ * @category	Class
+ * @author		WooThemes
+ */
+class WC_Payment_Tokens {
+
+	/**
+	 * Makes sure that the passed data is suitable for storing a token
+	 * @param  array $args  An array of arguments (can be meta or core token field) to validate
+	 * @return boolean      True if the passed data is valid
+	 */
+	public static function validate( $args ) {
+		if ( empty( $args['token'] ) ) {
+			return false;
+		}
+
+		if ( empty( $args['type'] ) ) {
+			return false;
+		}
+
+		// If 'token' and 'type' are present, validate that any specific token types (credit card for example)
+		// contain their neccesary fields.
+		$is_valid = true;
+		$token_class = 'WC_Payment_Token_' . $args['type'];
+		if ( is_callable( $token_class, 'validate' ) ) {
+			$is_valid = $token_class::validate( $args );
+		}
+
+		return apply_filters( 'woocommerce_payment_token_validate', $is_valid, $args );
+	}
+
+	/**
+	 * Creates, stores in the database, and returns a new payment token object
+	 * @param  array $args  An array of arguments (meta or core token fields)
+	 * @return WC_Payment_Token
+	 */
+	public static function create( $args ) {
+		if ( ! self::validate( $args ) ) {
+			return; // @todo throw an error
+		}
+
+		global $wpdb;
+		unset( $args['token_id'] );
+
+		// We need to separate our meta fields since they are stored in a separate table
+		$core_fields = self::get_token_core_fields();
+		$core_fields_to_create = array();
+		foreach ( $core_fields as $core_field ) {
+			if ( isset( $args[ $core_field ] ) ) {
+				$core_fields_to_create[ $core_field ] = $args[ $core_field ];
+				unset( $args[ $core_field ] );
+			}
+		}
```

Comment:
I thought about doing something similar to that, but most fields outside the token/gateway/customer are going to be extra so I thought it would make it nicer for devs to just pass in args without worrying. But I'm willing to implement either method depending on what people think is better. :smile:.